okio/src/test/java/okio/ReadUtf8LineTest.java

```diff
@@ -194,6 +194,55 @@
     assertEquals("ghi\rjkl", source.readUtf8LineStrict());
   }
 
+  @Test public void readUtf8LineWithLimits() throws IOException {
+    for (int limit : new int[] {
+        5, 6, 7, 10000
+    }) {
+      data.writeUtf8("panda\r\n");
+      assertEquals("panda", source.readUtf8Line(limit));
+      assertTrue(data.exhausted());
+    }
+
+    // newline not within limit
+    for (String input : new String[] {
+        "panda", "panda\n", "panda\r\n",
+    }) {
+      data.writeUtf8(input);
+      assertEquals("p", source.readUtf8Line(1));
+      assertEquals("an", source.readUtf8Line(2));
+      assertEquals("da", source.readUtf8Line(2));
+      assertTrue(data.exhausted());
+    }
+
+    // Trim trailing newlines
```

Comment:
This is good feedback! As you mention it, if one of these cases starts failing, it wouldn't be clear which input caused the problem.